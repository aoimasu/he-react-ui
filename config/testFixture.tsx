import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture = {
  component: React.ComponentType<any>;
  name?: string;
};

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
