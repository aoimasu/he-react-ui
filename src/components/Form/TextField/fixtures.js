// @flow
import React from 'react';
import Input from '.';

export default [
  {
    name: 'Default',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
    },
  },

  {
    name: 'small',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      small: true,
    },
  },

  {
    name: 'smallNoLabel',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      small: true,
    },
  },

  {
    name: 'password',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'Password',
      password: true,
    },
  },

  {
    name: 'noLabel',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
    },
  },

  {
    name: 'Disabled',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      disabled: true,
    },
  },

  {
    name: 'Error',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      error: "Something isn't right",
    },
  },

  {
    name: 'InvalidMarker',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      error: "Something isn't right",
      marker: true,
      value: 'Bruce',
    },
  },

  {
    name: 'ValidMarker',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      marker: true,
      value: 'Bruce',
    },
  },

  {
    name: 'Value',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      value: 'Oscar',
    },
  },

  {
    name: 'With helper',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      helper: <div>Test HTML</div>,
    },
  },

  {
    name: 'With verified',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      verified: <div>Verify</div>,
    },
  },
];
