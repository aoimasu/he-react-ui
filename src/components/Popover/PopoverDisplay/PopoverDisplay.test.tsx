import fixtures from './PopoverDisplay.fixtures';
import testFixture from '../../../../config/testFixture';

testFixture(fixtures);
