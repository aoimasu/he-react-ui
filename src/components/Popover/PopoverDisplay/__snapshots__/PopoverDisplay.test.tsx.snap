// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopoverDisplay /> rendered correctly with base fixture 1`] = `
<PopoverDisplay
  tooltip={true}
>
  <div
    className="popover tooltip"
    role="tooltip"
  >
    <div
      className="content tooltip"
    >
      I am a single-line tooltip
    </div>
  </div>
</PopoverDisplay>
`;

exports[`<PopoverDisplay /> rendered correctly with light fixture 1`] = `
<PopoverDisplay
  tooltip={true}
>
  <div
    className="popover tooltip"
    role="tooltip"
  >
    <div
      className="content tooltip"
    >
      I am a single-line tooltip
    </div>
  </div>
</PopoverDisplay>
`;

exports[`<PopoverDisplay /> rendered correctly with tooltip fixture 1`] = `
<PopoverDisplay
  tooltip={true}
>
  <div
    className="popover tooltip"
    role="tooltip"
  >
    <div
      className="content tooltip"
    >
      I am a single-line tooltip
    </div>
  </div>
</PopoverDisplay>
`;
